drivers/pds/lib/utils.go

```diff
@@ -481,11 +504,310 @@ func DeleteDeployment(deploymentID string) (*state.Response, error) {
+					Command: []string{"/bin/sh", "-c", command},
+					Env:     make([]corev1.EnvVar, 3),
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyOnFailure,
+		},
+	}
+
+	value = append(value, dnsEndpoint)
+	value = append(value, "pds")
+	value = append(value, pdsPassword)
+
+	for index := range env {
+		podSpec.Spec.Containers[0].Env[index].Name = env[index]
+		podSpec.Spec.Containers[0].Env[index].Value = value[index]
+	}
+
+	pod, err := k8sCore.CreatePod(podSpec)
+	if err != nil {
+		logrus.Errorf("An Error Occured while creating %v", err)
+		return nil, err
+	}
+
+	err = k8sCore.ValidatePod(pod, timeOut, timeInterval)
+	if err != nil {
+		logrus.Errorf("An Error Occured while validating the pod %v", err)
+		return nil, err
+	}
+
+	//TODO: Remove static sleep and verify the injected data
+	time.Sleep(1 * time.Minute)
+
+	return pod, nil
+}
+
+//CreateRmqWorkload generate workloads for rmq
+func CreateRmqWorkload(dnsEndpoint string, pdsPassword string, namespace string, env []string, command string) (*corev1.Pod, error) {
+	var value []string
+	podSpec := &corev1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: "rmq-perf-",
+			Namespace:    namespace,
+		},
+		Spec: corev1.PodSpec{
+			Containers: []corev1.Container{
+				{
+					Name:    "rmqperf",
+					Image:   rmqStressImage,
+					Command: []string{command},
+					Env:     make([]corev1.EnvVar, 3),
+				},
+			},
+			RestartPolicy: corev1.RestartPolicyOnFailure,
+		},
+	}
+
+	value = append(value, dnsEndpoint)
+	value = append(value, "pds")
+	value = append(value, pdsPassword)
+
+	for index := range env {
+		podSpec.Spec.Containers[0].Env[index].Name = env[index]
+		podSpec.Spec.Containers[0].Env[index].Value = value[index]
+	}
+
+	pod, err := k8sCore.CreatePod(podSpec)
+	if err != nil {
+		logrus.Errorf("An Error Occured while creating %v", err)
+		return nil, err
+	}
+
+	err = k8sCore.ValidatePod(pod, timeOut, timeInterval)
+	if err != nil {
+		logrus.Errorf("An Error Occured while validating the pod %v", err)
+		return nil, err
+	}
+
+	//TODO: Remove static sleep and verify the injected data
+	time.Sleep(1 * time.Minute)
+
+	return pod, nil
+}
+
+//CreateDataServiceWorkloads func
+func CreateDataServiceWorkloads(dataServiceName string, deploymentID string, scalefactor string, iterations string, deploymentName string, namespace string) (*corev1.Pod, *v1.Deployment, error) {
+	var dep *v1.Deployment
+	var pod *corev1.Pod
+
+	dnsEndpoint, err := GetDeploymentConnectionInfo(deploymentID)
+	if err != nil {
+		logrus.Errorf("An Error Occured while getting connection info %v", err)
+		return nil, nil, err
+	}
+	logrus.Infof("Dataservice DNS endpoint %s", dnsEndpoint)
+
+	pdsPassword, err := GetDeploymentCredentials(deploymentID)
+	if err != nil {
+		logrus.Errorf("An Error Occured while getting credentials info %v", err)
+		return nil, nil, err
+	}
+
+	switch dataServiceName {
+	case "PostgreSQL":
```

Comment:
make these as constants as I think you might use these dataservice names frequently

Reply:
done

Reply:
done